tfhe/src/high_level_api/integers/oprf.rs

```diff
@@ -0,0 +1,93 @@
+use super::{FheIntId, FheUintId};
+use crate::high_level_api::global_state;
+use crate::high_level_api::keys::InternalServerKey;
+use crate::integer::oprf::SignedRandomizationSpec;
+use crate::{FheInt, FheUint};
+use concrete_csprng::seeders::Seed;
+
+/// Generates an encrypted `num_block` blocks unsigned integer
+/// taken uniformly in `[0, 2^random_bits_count[` using the given seed
+/// The encryted value is oblivious to the server
+/// It can be useful to make server random generation deterministic
+///
+/// ```rust
+/// use tfhe::integer::gen_keys_radix;
+/// use tfhe::integer::oprf::SignedRandomizationSpec;
+/// use tfhe::prelude::FheDecrypt;
+/// use tfhe::{
+///     generate_keys, par_generate_oblivious_pseudo_random_unsigned_integer, set_server_key,
+///     ConfigBuilder, FheUint8, Seed,
+/// };
+///
+/// let config = ConfigBuilder::default().build();
+/// let (client_key, server_key) = generate_keys(config);
+///
+/// set_server_key(server_key);
+///
+/// let random_bits_count = 3;
+///
+/// let ct_res: FheUint8 =
+///     par_generate_oblivious_pseudo_random_unsigned_integer(Seed(0), random_bits_count);
+///
+/// let dec_result: u16 = ct_res.decrypt(&client_key);
+/// assert!(dec_result < (1 << random_bits_count));
+/// ```
+pub fn par_generate_oblivious_pseudo_random_unsigned_integer<Id: FheUintId>(
+    seed: Seed,
+    random_bits_count: u64,
```

Comment:
I wonder if we should have a function that just take the seed, and generates as many bits as the type has
by using `Id::num_bits()`

Also I wonder if these should not be associated function like `FheUint::generate_obliviopus_pseudo_random(seed)`
 `FheUint::generate_obliviopus_pseudo_random_bits(seed, num_bits)`

Reply:
I don't think it's very useful
The current API supports both cases and I don't think having `bits=precision` is a case special enough

Reply:
imo, the special case is  `Id::num_bits() != random_bit_count`, like we generally do `rng.gen::<u64>()` and not `rng.gen<u64>(u64::BITS)`

Reply:
I would tend to think that the associated function makes more sense, agree that generally the special case (at least from our perspective during implementation) is when the requested number of bits is not the one from the type

Reply:
Done